src/midi/midi_mt32.cpp

```diff
@@ -146,41 +238,17 @@ static std::deque<std::string> get_rom_dirs()
 
 #endif
 
-static bool load_rom_set(const std::string &ctr_path,
-                         const std::string &pcm_path,
-                         MidiHandler_mt32::service_t &service)
+static std::string load_model(const MidiHandler_mt32::service_t &service,
+                              const std::string &selected_model,
+                              const std::deque<std::string> &rom_dirs)
 {
-	const bool paths_exist = path_exists(ctr_path) && path_exists(pcm_path);
-	if (!paths_exist)
-		return false;
-
-	const bool roms_loaded = (service->addROMFile(ctr_path.c_str()) ==
-	                          MT32EMU_RC_ADDED_CONTROL_ROM) &&
-	                         (service->addROMFile(pcm_path.c_str()) ==
-	                          MT32EMU_RC_ADDED_PCM_ROM);
-	return roms_loaded;
-}
-
-static bool find_and_load(const std::string &model,
-                          const std::deque<std::string> &rom_dirs,
-                          MidiHandler_mt32::service_t &service)
-{
-	const std::string ctr_rom = model + "_CONTROL.ROM";
-	const std::string pcm_rom = model + "_PCM.ROM";
-	for (const auto &dir : rom_dirs) {
-		if (load_rom_set(dir + ctr_rom, dir + pcm_rom, service)) {
-			LOG_MSG("MT32: Found ROM pair in %s", dir.c_str());
-
-			mt32emu_rom_info rom_info;
-			service->getROMInfo(&rom_info);
-			LOG_MSG("MT32: Initialized %s with %s",
-			        rom_info.control_rom_description,
-			        rom_info.pcm_rom_description);
-
-			return true;
-		}
-	}
-	return false;
+	const bool is_auto = (selected_model == "auto");
+	for (auto &model : all_models)
```

Comment:
`const auto &` perhaps? :)

Reply:
I was caching the result of `Model.InDir("some/directory")` in the ``map<string> in_dir` member variable, to avoid repeat lookups (which are a tad expensive.. involves filesystem checks followed by up to 4 SHA1 hashes being run on the files).

Writing the cache to the Model's member meant I couldn't use `const` in these handful of areas. 

The reason I used the cache was because these InDir checks can repeat in the `mixer /listmidi`.  However.. realistically these are going to be rare events, and repeating a couple 32KB SHA1 passes may only take a could more milliseconds :sweat_smile:.

So I've dropped that complexity for cleaner `const` goodness :-)

Thanks for the suggestion!